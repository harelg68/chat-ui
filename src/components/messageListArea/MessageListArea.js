/**
 * Created by harelg on 30/01/2018.
 */

import React, {Component} from 'react';
import PropTypes from 'prop-types';
import {List, Container} from 'semantic-ui-react'
import Message from './Message';

class MessageListArea extends Component {
  render() {

    const {userName, messages} = this.props;

    return <Container className={'message-list-area'}>
        <List className={'message-list'}>
          {
            messages.map((message, inx) => {
              return <Message key={inx} message={message} isOwner={message.username === userName} />;
            })
          }
        </List>
      </Container>
  }
}

MessageListArea.propTypes = {
  userName: PropTypes.string.isRequired,
  messages: PropTypes.array.isRequired
};

export default MessageListArea;