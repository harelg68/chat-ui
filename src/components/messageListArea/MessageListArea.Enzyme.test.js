/**
 * Created by harelg on 02/02/2018.
 */
import React from 'react';
import { shallow, render } from 'enzyme';
import toJson from 'enzyme-to-json';

import Message from './Message';
import MessageListArea from './MessageListArea';

const props = {
  userName: 'Harel',
  messages: [{
    avatar: "dog.png",
    username: "Harel",
    text: "Test message...",
    timestamp: 1517510731694
  }]
};

// Testing <MessageListArea> component rendering.
describe('Testing <MessageListArea> component rendering', () => {
  it('should be rendered correctly', () => {
    const wrapper = render(<MessageListArea {...props} />);

    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should render no messages when given an empty list', () => {
    const wrapper = shallow(<MessageListArea userName='Harel' messages={[]} />);

    expect(wrapper.find(Message).length).toEqual(0);
  });

  it('should render the correct amount of messages', () => {
    const wrapper = shallow(<MessageListArea {...props} />);

    expect(wrapper.find(Message).length).toEqual(props.messages.length);
  });
});