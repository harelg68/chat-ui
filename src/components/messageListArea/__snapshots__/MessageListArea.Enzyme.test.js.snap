// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing <MessageListArea> component rendering should be rendered correctly 1`] = `
<div
  class="ui container message-list-area"
>
  <div
    class="ui list message-list"
    role="list"
  >
    <div
      class="item"
      role="listitem"
    >
      <div
        class="content"
      >
        <div
          class="clsMsgNormal"
        >
          <div
            class="clsMsgTime"
          >
            02/01/18 20:45:31
          </div>
          <img
            class="clsMsgIcon"
            src="dog.png"
          >
            
          </img>
          <div
            class="clsMsgItem clsOwnerMe"
          >
            Harel
          </div>
          <div
            class="clsMsgText"
          >
            Test message...
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
